content-moderation/03-audio-moderation/01-content-moderation-audio-transcribe-with-custom-vocabulary.ipynb

```diff
@@ -0,0 +1,341 @@
+    "\n",
+    "Open the \"my-vocabulary-filter.txt\" file in the datasets directory and see the list of words entered there. If you wish to add additional terms, please do so and save the file. Execute this cell afterwords. We then will upload it to the default S3 bucket for Transcribe to access."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Upload the text file with profane terms to the default S3 bucket\n",
+    "s3_vocabfile_key = 'content-moderation-im/audio-moderation/my-vocabulary-filters/my-vocabulary-filter.txt'\n",
+    "s3.upload_file('../datasets/my-vocabulary-filter.txt', data_bucket, s3_vocabfile_key)\n",
+    "vocab_file_uri = 's3://'+data_bucket+'/'+s3_vocabfile_key\n",
+    "print(vocab_file_uri)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's call Transcribe **CreateVocabularyFilter** API to create a new vocabulary filter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Change the name for vocab_name if you have created a custom vocabulary already\n",
+    "vocab_name = \"audio-moderation-filter\"\n",
+    "response = transcribe_client.create_vocabulary_filter(\n",
+    "    LanguageCode = 'en-US',\n",
+    "    VocabularyFilterName = vocab_name,\n",
+    "    VocabularyFilterFileUri = vocab_file_uri\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 5: Run transcribe Job with Custom Vocabulary Filter <a id=\"step5\"></a>\n",
+    "In this step, we will call the same Transcribe `StartTranscriptionJob` API but pass the Vocabulary Filter created in the previous step as an additional parameter under `Settings`. So Transcribe will transcribe the audio to text and apply the profane filter and masking based on the setting."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#In this case we are running the transcribe job with option to \"mask\" the words. You can also run the job with \"remove\" or \"tag\". \n",
+    "#In each case, give a new job_name and change the VocabularyFilterMethod to the desired option\n",
+    "\n",
+    "job_name = \"transcription_with_audio_moderation_mask1\"\n",
+    "transcribe_client.start_transcription_job(\n",
+    "    TranscriptionJobName = job_name,\n",
+    "    Media = {\n",
+    "        'MediaFileUri': file_uri\n",
+    "    },\n",
+    "    OutputBucketName = data_bucket,\n",
+    "    OutputKey = 'content-moderation-im/audio-moderation/my-output-files/', \n",
+    "    LanguageCode = 'en-US', \n",
+    "    Settings = {\n",
+    "        'VocabularyFilterName': vocab_name,\n",
+    "        'VocabularyFilterMethod': 'mask' \n",
+    "   }\n",
+    ")\n",
+    "\n",
+    "while True:\n",
```

Comment:
This is a bit inconsistent with before, where you implemented max retries, here instead you keep going forever.
My preference would be to do this consistently (either implement max retries everywhere or not).

What do you think?

Reply:
The AI API has built-in retry. The code heart-beating the API to pull the job status. This lab may have a different style than the others, but it should work well for the lab.